lib/utilities/firebase_service.dart

```diff
@@ -108,108 +130,274 @@ class FirebaseService {
+        final eventSnapshot = await eventDocRef.get();
+        if (!eventSnapshot.exists) return null;
+
+        final data = eventSnapshot.data() as Map<String, dynamic>;
+        data['id'] = eventSnapshot.id;
+
+        // Check if the event is a past event
+        final eventDateTime = (data['dateTime'] as Timestamp).toDate();
+        final isPastEvent = eventDateTime.isBefore(DateTime.now());
+        if (isPastEvent) {
+          return SportEvent.fromMap(data);
+        }
+        return null;
+      }),
+    );
+
+    // Filter out null results and return the list of SportEvent
+    return pastEvents.whereType<SportEvent>().toList();
+  } catch (e) {
+    print('Error fetching past events: $e');
+    return [];
+  }
+}
 
-    // Fetch all events and filter manually for past events
-    final eventsSnapshot = await _firestore.collection('events-collection').get();
+  Future<List<SportEvent>> getHostedEvents(String userId) async {
+  try {
+    // Query the 'host' collection to get all event IDs where the user is the host
+    final hostSnapshot = await _firestore
+        .collection('host')
+        .where('userId', isEqualTo: _firestore.collection('users').doc(userId)) // Match userId as DocumentReference
+        .get();
 
-    final pastEvents = eventsSnapshot.docs.where((doc) {
-      final data = doc.data() as Map<String, dynamic>;
-      final eventDateTime = data['dateTime'] as Timestamp;
-      final isPastEvent = eventDateTime.toDate().isBefore(DateTime.now());
-      return eventIds.contains(doc.id) && isPastEvent;
-    }).toList();
+    // Extract event IDs from the 'host' collection
+    final eventIds = hostSnapshot.docs.map((doc) {
+      final data = doc.data();
+      final eventRef = data['eventId'] as DocumentReference?;
+      return eventRef?.id; // Get the ID of the DocumentReference
+    }).whereType<String>().toList(); // Remove nulls
 
-    return pastEvents.map((doc) {
-      final data = doc.data() as Map<String, dynamic>;
-      data['id'] = doc.id;
+    if (eventIds.isEmpty) {
+      return []; // No hosted events found
+    }
+
+    // Query the 'events-collection' for events matching the retrieved IDs
+    final eventSnapshot = await _firestore
+        .collection('events-collection')
+        .where(FieldPath.documentId, whereIn: eventIds) // Use event IDs
+        .get();
+
+    // Map the results to a list of SportEvent
+    return eventSnapshot.docs.map((doc) {
+      final data = doc.data();
+      data['id'] = doc.id; // Include the document ID
       return SportEvent.fromMap(data);
     }).toList();
   } catch (e) {
-    print('Error fetching past events: $e');
+    print('Error fetching hosted events: $e');
     return [];
   }
 }
+Future<List<String>> getUserNames(List<String> userIds) async {
+  try {
+    List<String> userNames = [];
+
+    for (String userId in userIds) {
+      final userDoc = await _firestore.collection('users').doc(userId).get();
+      if (userDoc.exists) {
```

Comment:
This is a bad design. Doing a database query for each item in the list of userIds is bad. What happens when there's a million userIds? 
You should only make one query and get all the users at once with a contains query.